plugin/pkg/shoot/validator/admission.go

```diff
@@ -365,7 +365,29 @@ func (c *validationContext) validateScheduling(a admission.Attributes, shootList
 		}
 
 		if oldSeed.Spec.Provider.Type != c.seed.Spec.Provider.Type {
-			return admission.NewForbidden(a, fmt.Errorf("cannot change Seed because cloud provider for new seed (%s) is not equal to cloud provider for old seed (%s)", c.seed.Spec.Provider.Type, oldSeed.Spec.Provider.Type))
+			return admission.NewForbidden(a, fmt.Errorf("cannot change seed because cloud provider for new seed (%s) is not equal to cloud provider for old seed (%s)", c.seed.Spec.Provider.Type, oldSeed.Spec.Provider.Type))
+		}
+
+		if c.shoot.Status.LastOperation != nil && c.shoot.Status.LastOperation.State != core.LastOperationStateSucceeded {
+			return admission.NewForbidden(a, fmt.Errorf("cannot change seed name because the last operation has not finished successfully yet: operation %s, state %s", c.shoot.Status.LastOperation.Type, c.shoot.Status.LastOperation.State))
+		}
+
+		if isShootInMigrationOrRestorePhase(c.shoot) && !reflect.DeepEqual(c.oldShoot.Spec, c.shoot.Spec) {
+			return admission.NewForbidden(a, fmt.Errorf("cannot change shoot spec during %s operation that is in state %s", c.shoot.Status.LastOperation.Type, c.shoot.Status.LastOperation.State))
+		}
+
+		if err := func() error {
+			oldSeedName := c.oldShoot.Spec.SeedName
+			c.oldShoot.Spec.SeedName = c.shoot.Spec.SeedName
+			defer func() {
+				c.oldShoot.Spec.SeedName = oldSeedName
+			}()
```

Comment:
Instead of such construct, can you rather make copies of the shoots and modify them according to your needs?